### PR TITLE
[cluster-autoscaler] chore: Rename Cluster Autoscaler Provider package from Packet to EquinixMetal

### LINKED ISSUE #4286: Rename all references of Packet to Equinix Metal
Which component hosted in this repository is this a feature for? cluster-autoscaler

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

Since Equinix's acquisition of Packet, other repositories and projects are being renamed from Packet to Equinix Metal in code, config and documentation.

I'd like this to happen for `cluster-autoscaler/cloudprovider/packet` as well.

**Describe the solution you'd like.**:

All references are correctly renamed from Packet to Equinix Metal.

### PR DESCRIPTION
**Pull request by aayushrangwala**
#### What type of PR is this?

/kind cleanup
/kind deprecation

#### What this PR does / why we need it:

This PR will rename the packages and file names from Packet to equinixmetal

This is a sequential PR depends on https://github.com/kubernetes/autoscaler/pull/6084 and https://github.com/kubernetes/autoscaler/pull/6085

#### Which issue(s) this PR fixes:

Fixes #4286

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/builder/builder_packet.go [hunk 1]
```diff
@@ -21,23 +21,25 @@ package builder
...
 // DefaultCloudProvider for Packet-only build is Packet.
 const DefaultCloudProvider = cloudprovider.EquinixMetalProviderName
 
 func buildCloudProvider(opts config.AutoscalingOptions, do cloudprovider.NodeGroupDiscoveryOptions, rl *cloudprovider.ResourceLimiter) cloudprovider.CloudProvider {
 	switch opts.CloudProviderName {
-	case packet.ProviderName, cloudprovider.EquinixMetalProviderName:
-		return packet.BuildCloudProvider(opts, do, rl)
+	case equinixmetal.ProviderName, cloudprovider.EquinixMetalProviderName:
+		return equinixmetal.BuildCloudProvider(opts, do, rl)
+	case cloudprovider.EquinixMetalProviderName:
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/builder/builder_packet.go:41]: This case duplicates the case above

File: cluster-autoscaler/cloudprovider/builder/builder_packet.go [hunk 2]
```diff
@@ -21,23 +21,25 @@ package builder
...
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider/equinixmetal"
 	"k8s.io/autoscaler/cluster-autoscaler/config"
 )
 
 // AvailableCloudProviders supported by the cloud provider builder.
 var AvailableCloudProviders = []string{
-	packet.ProviderName,
-	cloudprovider.EquinixMetalProviderName,
+	equinixmetal.ProviderName,
+	equinixmetal.EquinixMetalProviderName,
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/builder/builder_packet.go:31]: This use of equinixmetal.EquinixMetalProviderName is inconsistent with line 42 below.

Did you want:
```suggestion
	cloudprovider.EquinixMetalProviderName,
```

File: cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go [hunk 3]
```diff
@@ -578,7 +578,7 @@ func (mgr *equinixMetalManagerRest) createDeviceRequest(ctx context.Context, cr
 // getNodes should return ProviderIDs for all nodes in the node group,
 // used to find any nodes which are unregistered in kubernetes.
 func (mgr *equinixMetalManagerRest) getNodes(nodegroup string) ([]string, error) {
-	// Get node ProviderIDs by getting device IDs from Packet
+	// Get node ProviderIDs by getting device IDs from metal API
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go:581]:
```suggestion
	// Get node ProviderIDs by getting device IDs from the Equinix Metal API
```

File: cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ type instanceType struct {
 	GPU          int64
 }
 
-// InstanceTypes is a map of packet resources
+// InstanceTypes is a map of metal resources
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go:61]:
```suggestion
// InstanceTypes is a map of equinix metal resources
```

`metal` alone shouldn't be used to describe this provider in a multi-provider solution. There are other baremetal cloud provider services that use this common service name.

File: cluster-autoscaler/cloudprovider/equinixmetal/cloud_provider.go [hunk 5]
```diff
@@ -35,7 +35,7 @@ import (
 )
 
 const (
-	// ProviderName is the cloud provider name for Packet cloud provider, now named as equinixmetal
+	// ProviderName is the cloud provider name for Equinix Metal
 	ProviderName = "packet"
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/equinixmetal/cloud_provider.go:39]: I'm confused by the use of cloudprovider.EquinixMetalProviderName and ProviderName (packet) being used in this package.

I think `ProviderName` should be removed from here entirely. All references can be updated to `cloudprovider.PacketProviderName`

File: cluster-autoscaler/cloudprovider/builder/builder_packet.go (hunk 1 above)
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/builder/builder_packet.go:41]: Ah right, missed this while rebasing

File: cluster-autoscaler/cloudprovider/builder/builder_packet.go (hunk 2 above)
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/builder/builder_packet.go:31]: Yes @displague meant to use `cloudprovider.EquinixMetalProviderName`. Fixed

Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/builder/builder_packet.go:31]: @Shubham82 Yes, correct. Good find. Fixed it

File: cluster-autoscaler/cloudprovider/builder/builder_all.go [hunk 6]
```diff
@@ -122,7 +122,7 @@ func buildCloudProvider(opts config.AutoscalingOptions, do cloudprovider.NodeGro
 	case cloudprovider.HetznerProviderName:
 		return hetzner.BuildHetzner(opts, do, rl)
 	case cloudprovider.PacketProviderName, cloudprovider.EquinixMetalProviderName:
-		return packet.BuildCloudProvider(opts, do, rl)
+		return equinixmetal.BuildCloudProvider(opts, do, rl)
```
[the remark this answers is not in the document]
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:125]: Yes fixed

File: cluster-autoscaler/cloudprovider/equinixmetal/cloud_provider.go (hunk 5 above)
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/equinixmetal/cloud_provider.go:39]: Yes, it was redundant before by having provider name at 2 places. Only using one variable  `cloudprovider.PacketProviderName`

File: cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go (hunk 4 above)
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go:61]: Yes right. Fixed

File: cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go (hunk 3 above)
Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/equinixmetal/manager_rest.go:581]: Fixed

**Comment by Shubham82** [comment]
@aayushrangwala, The changes look good to me.
Let's wait for @displague comment.

Thanks!

File: cluster-autoscaler/cloudprovider/builder/builder_all.go [hunk 7]
```diff
@@ -75,7 +75,7 @@ var AvailableCloudProviders = []string{
 	cloudprovider.LinodeProviderName,
 	cloudprovider.BizflyCloudProviderName,
 	cloudprovider.BrightboxProviderName,
-	cloudprovider.PacketProviderName,
```
Comment (displague) [inline cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: I think we still need PacketProviderName here for the provider to legacy provider to be used

Comment (displague) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: I see - AvailableCloudProviders is only used for command line documentation, it is not used for command line validation.  This should be fine then, as it directs users to use the new provider name but does not prevent usage of the legacy name.

**Review by displague — APPROVED** [review]

File: cluster-autoscaler/cloudprovider/builder/builder_all.go (hunk 7 above)
Comment (Shubham82) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: I think we have to change the name here also, to maintain uniformity.

Comment (aayushrangwala) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: Yes, already changed and added equinix provider name `cloudprovider.EquinixMetalProviderName` on line 78

**Comment by aayushrangwala** [comment]
@feiskyer Can you please approve this PR as suggested by the k8s bot https://github.com/kubernetes/autoscaler/pull/6271#issuecomment-1816219469

File: cluster-autoscaler/cloudprovider/builder/builder_all.go (hunk 7 above)
Comment (Shubham82) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: Thanks @aayushrangwala

Comment (Shubham82) [reply cluster-autoscaler/cloudprovider/builder/builder_all.go:78]: looks good to me.
/lgtm

**Comment by Shubham82** [comment]
@aayushrangwala please rebase the PR, so that merge conflicts will be resolved.

**Comment by aayushrangwala** [comment]
> @aayushrangwala please rebase the PR, so that merge conflicts will be resolved.

Rebased

**Comment by aayushrangwala** [comment]
Hey @blind3dd I had to rebase the PR, can you please approve again

**Review by feiskyer — APPROVED** [review]
LGTM

**Pull request merged**